augur/filter.py

```diff
@@ -105,6 +105,8 @@ def register_arguments(parser):
                                 help="Exclude samples matching these conditions. Ex: \"host=rat\" or \"host!=rat\". Multiple values are processed as OR (matching any of those specified will be excluded), not AND")
     parser.add_argument('--include-where', nargs='+',
                                 help="Include samples with these values. ex: host=rat. Multiple values are processed as OR (having any of those specified will be included), not AND. This rule is applied last and ensures any sequences matching these rules will be included.")
+    parser.add_argument('--exclude-ambiguous-dates', choices=['all', 'days', 'months'],
```

Comment:
Having thought about this more and seeing this actual implementation, I would slightly prefer the argument here to be `--exclude-ambiguous-dates-by` and the argument values to be `day`, `month`, `year`, or `all`.

This is a minor preference for readability and I'm OK with keeping the current name if @jameshadfield or @emmahodcroft think the above suggestion is too verbose or less clear.

Either way, it seems like we should include "year" for completeness.